autorest/date/time.go

```diff
@@ -19,7 +20,8 @@ func ParseTime(date string) (d Time, err error) {
 }
 
 func parseTime(date string, format string) (Time, error) {
-	d, err := time.Parse(format, date)
+	// Note: Convert the date first to uppercase since RFC3339 allows lower-case "t" and "z".
+	d, err := time.Parse(format, strings.ToUpper(date))
```

Comment:
We need to move ToUpper() to UnmarsalJSON() because we are using that function now. 

```go
func (d *Time) UnmarshalJSON(data []byte) (err error) {
        return d.Time.UnmarshalJSON(bytes.ToUpper(data))
 }
```